ecc/enclave_chaincode.go

```diff
@@ -174,6 +167,134 @@ func (t *EnclaveChaincode) setup(stub shim.ChaincodeStubInterface) pb.Response {
 	return shim.Success([]byte(enclavePkBase64))
 }
 
+// ============================================================
+// createenclave -
+// ============================================================
+func (t *EnclaveChaincode) createenclave(stub shim.ChaincodeStubInterface) pb.Response {
```

Comment:
all done.

Reply:
Actually, you changed too much: the external mgmt api is called `createenclave` (which is then implemented internally via an initEnclave call to ecc and a `registerEnclave` to ercc)

Reply:
> Oh, i see you also changed the spec to uniformally change everywhere to initenclave (rather than the distinction from before).

well, it does not make much sense at this point to have a `createenclave` that magically becomes `initenclave`.

> PS: seems you also missed a few createenclave in error messages in ecc/enclave_chaincode.go

ouch!

> In that case, you also will have to change the client sdk ; although that might give marcus some grief as he has related changes in his WIP PR. Not changing the specs might have been easier but i have no strong opinion on name, you and marcus have to figure out why way to resolve ...

This brings to an observation and a request.
Observation: from my perspective, this renaming would bring grief to any developer -- including this PR.

Request (@g2flyer @mbrandenburger ):
as _any_ renaming does not change functionality, but _might_ clearly impact other developers, would you be ok to confine any such renaming to a separate "renaming PR", which does not modify any functionality?

Reply:
> Request (@g2flyer @mbrandenburger ):
> as _any_ renaming does not change functionality, but _might_ clearly impact other developers, would you be ok to confine any such renaming to a separate "renaming PR", which does not modify any functionality?

Hmm, you realize thought that _you_ did the renaming in this PR? Marcus was following what the spec did before your PR ....

Reply:
Yes, I do -- because I have kindly addressed the requests.
It'll bring even more grief if I have to revert it, but so it be if this solves a bigger problem.

Reply:
> Yes, I do -- because I have kindly addressed the requests.

But my request is to match initEnclave where the spec said initEnclave, not rename createenclave to initEnclave in the cli?

> It'll bring even more grief if I have to revert it, but so it be if this solves a bigger problem.

I guess the easiest is now that Marcus does the rename to from the `createenclave` admin cmd to `initenclave` admin cmd in his wip pr.  Above i just pointed out that client sdk has to be also changed and that you and marcus have to handshake on it; anyway, probably easiest to resolve tomorrow on call unless marcus just goes ahead and changes his WIP PR ....